### ISSUE TITLE
Soft Shadow Artifacts

### ISSUE THREAD

**Issue by Blitzoreo** [MadDeCoDeR/Classic-RBDOOM-3-BFG#26]
It seems that when you have Soft Shadows enabled, you may stumble upon many "shadow artifacts", such as flickering, noise/blur filtering and the like.
Disabling Soft Shadows fixes this, but not a good "fix" as you can tell.

The most prominent issue I have is the noise filtering, flickering (although luckily in this case it is very, very subtle) and 'fake shadows', as seen here:

LOD 0.4
![20190422202608_1](https://user-images.githubusercontent.com/4991692/56519470-6fd52f80-6542-11e9-9f81-c0079c5c044a.jpg)
LOD 2.0
![20190422202611_1](https://user-images.githubusercontent.com/4991692/56519477-72d02000-6542-11e9-95a2-522d3a860edb.jpg)
Soft Shadows OFF
![20190422202620_1](https://user-images.githubusercontent.com/4991692/56519484-75cb1080-6542-11e9-8f01-076e8a2151a4.jpg)

This seems to affect a few select textures, I haven't tried out.
Can this be fixed or at the very least improved?

**Comment by MadDeCoDeR** [comment]
Tbh I am not that familiar with how Soft Shadows work on this port. But I will see what I can do (Don't get too many hopes about it)

**Comment by RinMaru** [comment]
cant say ive experienced this issue might be safe to close till someone comes across it again

**Comment by Hexaae** [comment]
Found some places with heavy glitches when Soft Shadows is on:
https://youtu.be/KIuH9KD6zH0

**Comment by MadDeCoDeR** [comment]
Try disabling some post proccessing effects like SSAO and fimic effect.

**Comment by Hexaae** [comment]
> Try disabling some post proccessing effects like SSAO and fimic effect.

Tried. Won't change the glitches.

**Comment by CaptainCoward** [comment]
Even when soft shadows are "off" you'll have different artifacts. I haven't manage to get rid of it completly so far. Sometimes it is even worse with it set to "off".

**Comment by MadDeCoDeR** [comment]
Do you have HDR active?

**Comment by CaptainCoward** [comment]
No. HDR is off, Filmic Effect and SSAO are OFF as well. I tried different combinations. It's either OPs issue, slightly better, or even worse. So far best results i had was with SSAO "off "and Softshadows "on". 
Turning Softshadows off will sometimes casue shadows to have some strange "Z-Fight" going on.

**Comment by MadDeCoDeR** [comment]
Where exactly you have issue with the stencil shadows?

**Comment by QcSamuel** [comment]
any news on this issue ? i have film grain like artifacts

**Comment by MadDeCoDeR** [comment]
Sadly, no. To be honest I am embarrassingly noob when it come to graphics and I still don't even know what might causing the issue

**Comment by QcSamuel** [comment]
does it happen on RBDOOM too or is it a bfa issue ?

**Comment by MadDeCoDeR** [comment]
Even if it did or not, as far as I know RBDOOM have made some rather drastic changes to the rendering code that I can't follow (mainly because of Thy Warnings Consumed (treat warnings as errors)

**Comment by QcSamuel** [comment]
https://github.com/RobertBeckebans/RBDOOM-3-BFG/issues/237

**Comment by MadDeCoDeR** [comment]
Just adjusted jitter scale, issue looks to be gone and so some shadow details. Check the latest nightly builds if you want

**Comment by MadDeCoDeR** [comment]
I think I'm going in the right direction. The latest nightly release no longer has those artifacts
